src/support_enumeration.jl

```diff
@@ -0,0 +1,293 @@
+    n_min = min(nums_actions...)
+    S = typeof(zero(T)/one(T))
+    if S != T
+        payoff_matrices = NTuple{2, Matrix{S}}(payoff_matrices)
+    end
+
+    for k = 1:n_min
+        supps = (collect(1:k), Vector{Int}(k))
+        actions = (Vector{S}(k), Vector{S}(k))
+        A = Matrix{S}(k+1, k+1)
+        b = Vector{S}(k+1)
+        while supps[1][end] < nums_actions[1]+1
+            supps[2][:] = collect(1:k)
+            while supps[2][end] < nums_actions[2]+1
+                if _indiff_mixed_action!(A, b, actions[2],
+                                         payoff_matrices[1],
+                                         supps[1], supps[2])
+                    if _indiff_mixed_action!(A, b, actions[1],
+                                             payoff_matrices[2],
+                                             supps[2], supps[1])
+                        out = (zeros(S, nums_actions[1]),
+                               zeros(S, nums_actions[2]))
+                        for (p, (supp, action)) in enumerate(zip(supps,
+                                                                 actions))
+                            out[p][supp] = action
+                        end
+                        produce(out)
+                    end
+                end
+                _next_k_array!(supps[2])
+            end
+            _next_k_array!(supps[1])
+        end
+    end
+
+end
+
+"""
+    _indiff_mixed_action!{T<:Real}(A::Matrix{T}, b::Vector{T},
+                                   out::Vector{T},
+                                   payoff_matrix::Matrix{T},
+                                   own_supp::Vector{Int},
+                                   opp_supp::Vector{Int})
+
+Given a player's payoff matrix `payoff_matrix`, an array `own_supp`
+of this player's actions, and an array `opp_supp` of the opponent's
+actions, each of length k, compute the opponent's mixed action whose
+support equals `opp_supp` and for which the player is indifferent
+among the actions in `own_supp`, if any such exists. Return `true`
+if such a mixed action exists and actions in `own_supp` are indeed
+best responses to it, in which case the outcome is stored in `out`;
+`false` otherwise. Arrays `A` and `b` are used in intermediate
+steps.
+
+# Arguments
+
+* `A::Matrix{T}`: Matrix used in intermediate steps.
+* `b::Vector{T}`: Vector used in intermediate steps.
+* `out::Vector{T}`: Vector to store the nonzero values of the
+  desired mixed action.
+* `payoff_matrix::Matrix{T}`: The player's payoff matrix.
+* `own_supp::Vector{Int}`: Vector containing the player's action indices.
+* `opp_supp::Vector{Int}`: Vector containing the opponent's action indices.
+
+# Returns
+
+* `::Bool`: `true` if a desired mixed action exists and `false` otherwise.
+"""
+function _indiff_mixed_action!{T<:Real}(A::Matrix{T}, b::Vector{T},
+                                        out::Vector{T},
+                                        payoff_matrix::Matrix{T},
```

Comment:
Yes, it would be automatically converted to be the same with the eltype of `A`.

I am doing this because the original eltype of `payoff_matrix` is not going to be used anymore, so just drop it might be more concise. And rather than automatically converting the element of `payoff_matrix` a lot of time, doing it once [here](https://github.com/QuantEcon/Games.jl/pull/26#discussion_r109300381) might be more efficient, as I thought.